ConsumerApi.Tests.Integration/StepDefinitions/MessagesStepDefinitions.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System.Text.Json;
+using Backbone.BuildingBlocks.SDK.Endpoints.Common.Types;
+using Backbone.ConsumerApi.Sdk;
+using Backbone.ConsumerApi.Sdk.Authentication;
+using Backbone.ConsumerApi.Sdk.Endpoints.Messages.Types.Requests;
+using Backbone.ConsumerApi.Sdk.Endpoints.Messages.Types.Responses;
+using Backbone.ConsumerApi.Sdk.Endpoints.Relationships.Types.Requests;
+using Backbone.ConsumerApi.Sdk.Endpoints.RelationshipTemplates.Types.Requests;
+using Backbone.ConsumerApi.Tests.Integration.Configuration;
+using Backbone.ConsumerApi.Tests.Integration.Extensions;
+using Backbone.ConsumerApi.Tests.Integration.Support;
+using Backbone.Crypto;
+using Microsoft.Extensions.Options;
+
+namespace Backbone.ConsumerApi.Tests.Integration.StepDefinitions;
+
+[Binding]
+[Scope(Feature = "POST Message")]
+internal class MessagesStepDefinitions
+{
+    private Client _sdk1 = null!;
+    private Client _sdk2 = null!;
+    private ApiResponse<SendMessageResponse>? _sendMessageResponse;
+    private readonly ClientCredentials _clientCredentials;
+    private readonly HttpClient _httpClient;
+
+    public MessagesStepDefinitions(HttpClientFactory factory, IOptions<HttpConfiguration> httpConfiguration)
+    {
+        _httpClient = factory.CreateClient();
+        _clientCredentials = new ClientCredentials(httpConfiguration.Value.ClientCredentials.ClientId, httpConfiguration.Value.ClientCredentials.ClientSecret);
+    }
+
+    [Given("Identities i1 and i2 with an established Relationship")]
+    public async Task GivenIdentitiesI1AndI2WithAnEstablishedRelationship()
+    {
+        _sdk1 = await Client.CreateForNewIdentity(_httpClient, _clientCredentials, Constants.DEVICE_PASSWORD);
+        _sdk2 = await Client.CreateForNewIdentity(_httpClient, _clientCredentials, Constants.DEVICE_PASSWORD);
```

Comment:
You say `Client.CreateForNewIdentity` and save the result in a variable called `sdk` of type `Client`. That's quite inconsistent. You should rather call the variable `client`